### PR TITLE
Breaking: Drop Node 10 and upgrade Ember

### PR DESCRIPTION
**Pull request by mansona**
This is a follow-on commit from https://github.com/ember-learn/ember-styleguide/pull/412 and is in a similar style. This gets the ember-blog template ready for Ember 4.0 and removes some of the bugs that were present in the end of the 3.x cycle.

This is quite a large PR and is best reviewed commit-by-commit. If we think it's too large then I can probably split this one up into 10+ PRs 😱 I'm happy either way so let me know what you think 👍 

Also I mentioned this in Discord but there is actually a visual diff with this PR. As a result of the updated ember-styleguide we have now fixed the issue with single tick code blocks (like `this one`) not having any style. This seems to add quite a lot of code blocks to the release posts so we need to make sure that we're ok with this before we proceed 👍 I think you should be able to see the diff in question if you follow [this link](https://percy.io/Ember/empress-blog-ember-template/builds/14790323/changed/835882032?browser=chrome&browser_ids=19%2C20&subcategories=approved&viewLayout=side-by-side&viewMode=new&width=1280&widths=375%2C1280)

### CODE REVIEW

File: addon/components/ember-times-header.hbs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+{{! template-lint-disable no-unknown-arguments-for-builtin-components }}
 {{#if this.isNewsletter}}
     <LinkTo @route="tag" @model="Newsletter" class="blog-header" @aria-label="Read more issues of The Ember Times">
```
Comment (MelSumner) [inline addon/components/ember-times-header.hbs:3]: Question- do you need `@aria-label` or can you just use `aria-label`?

**Comment by MelSumner** [comment]
@mansona do we need all of these checks to pass?

**Review by MelSumner — APPROVED** [review]
I like the code block change, it makes it more readable. We should check the color contrast just to make sure (if we don't already have checkers in place). 

I left a couple of questions inline but overall it looks good. 👍

**Comment by mansona** [comment]
> @mansona do we need all of these checks to pass?

No we don't, this already improves the situation amazingly since the [last build on master](https://github.com/ember-learn/empress-blog-ember-template/actions/runs/1627639287)

**Comment by mansona** [comment]
> We should check the color contrast just to make sure (if we don't already have checkers in place).

If I remember correctly we checked it when we implemented it originally, and if we're not too happy with it then we can update it upstream 👍

File: addon/components/ember-times-header.hbs (hunk 1 above)
Comment (mansona) [reply addon/components/ember-times-header.hbs:3]: so this isn't something that's changed as part of this PR, What we should do is get this merged and then fix as many issues that are ignored using lint-to-the-future after this is merged so we don't let this PR spiral any further out of control 🙈

**Pull request merged**